nexus/post.ts

```diff
@@ -424,5 +424,119 @@ schema.extendType({
         })
       },
     })
+
+    t.field('deletePost', {
+      type: 'Post',
+      args: {
+        postId: schema.intArg({ required: true })
+      },
+      resolve: async (_parent, args, ctx) => {
+        const { postId } = args
+        const { userId } = ctx.request
+        if (!userId) throw new NotAuthorizedError()
+
+        const post = await ctx.db.post.findOne({
+          where: {
+            id: postId,
+          },
+          select: {
+            id: true,
+            authorId: true,
+          },
+        })
+
+        if (!post) throw new Error('Post not found.');
+
+        const currentUser = await ctx.db.user.findOne({
+          where: {
+            id: userId,
+          },
+        })
+
+        if (!currentUser) {
+          throw new Error('User not found.')
+        }
+
+        hasAuthorPermissions(post, currentUser)
+
+        const deleteFirstPhasePromises = [
+          ctx.db.comment.deleteMany({
```

Comment:
I think "comment thanks" (that is thanks given to comments within threads) need to be deleted before you can delete comments, unfortunately. It's lousy that we don't have cascading deletes.